src/main/kotlin/com/atlassian/performance/tools/virtualusers/LoadSegment.kt

```diff
@@ -0,0 +1,21 @@
+package com.atlassian.performance.tools.virtualusers
+
+import com.atlassian.performance.tools.virtualusers.api.browsers.CloseableRemoteWebDriver
+import java.io.BufferedWriter
+import java.util.*
+import java.util.concurrent.atomic.AtomicBoolean
+
+class LoadSegment(
+    val driver: CloseableRemoteWebDriver,
+    val output: BufferedWriter,
+    val done: AtomicBoolean,
+    val id: UUID,
+    val index: Int
+) : AutoCloseable {
+
+    override fun close() {
+        done.set(true)
+        output.close()
+        driver.close()
```

Comment:
Should we propagate exceptions thrown during the close?

Reply:
Yes. If closing fails, we want to know why.
We care about `output.close` to flush buffers so that all produced action metrics will be persisted on disk.
We care about `driver.close` so that users won't leak browser processes, when used locally.